setup.py

```diff
@@ -104,6 +104,30 @@ def _walk_develop_plugins(self):
             walk_plugins_setup_py(action=["develop"] + self.action_options,
                                   action_name=self.action_name)
 
+    def _fetch_eggs(self):
+        """This method it will fetch eggs and register into the cache.
+
+        Those will be necessary to deploy avocado inside a container.
+        """
+        # TODO: We need to avoid this hard coded version.
+        py_version = "3.10"
```

Comment:
I was really liking where this was going, but this makes it clear that the Python version will depend on the VM/container image or system that is used by the spawner.

The question becomes: why should we download/register just one version instead of all versions?

Second, why do we need to register it by name, instead of simply using the fully qualified URL, always?  I don't think other modules (such as the spawner plugins) will contain the literal formatting string for the Python version it wants... IMO it will always use something that will give it the result.  Something like:

```python

def get_egg_url(avocado_version=None, python_version=None):
   if avocado_version is None:
      avocado_version = VERSION_FROM_SETUPTOOLS
   if python_version is None:
      python_version = sys.version_info

   asset_name = f"avocado_framework-{avocado_version}-py{python_version}.egg   
   return = f"https://github.com/avocado-framework/avocado/releases/download/{avocado_version}/{asset_name}"

def get_egg(avocado_version=None, python_version=None):
   return asset.fetch(get_egg_url(avocado_version, python_version))

```

Reply:
@clebergnu, I had the same impression that we could also download multiple versions. But I left here the TODO comment, just get options here. At worst case scenario we could have assets being downloaded that are not used. But we can give a try and see if this will see a problem in the future.

With regards your example, where came the `asset` object (last line)?

The only `fetch()` method I found, is part of the `Asset` class, and has only timeout as argument.

Reply:
@beraldoleal I see two possibilities here:

I. Every single install of avocado will attempt to download another 4 (or 5) "avocados" (the per-version egg)
II. Avocado installs won't download extra eggs

The first scenario makes a lot of sense, having everything ready to deploy avocado, if you consider that Avocado will be managing container/vm creation to run tests.

But, we can not forget an also common (and currently the only) use case, which is the deployment of those environments being handled at another layer, usually a CI framework/tool.  With scenario one here, a container prepared with `pip install avocado-framework` will wastefully download 4-5x more "Avocados" then it needs.  Also, I'm very doubtful that RPM packaging rules will allow for this duplication of data... especially because those (eggs) would have to be built during the RPM build phase.

IMO we should start with making it possible to download some/all the eggs, then maybe making it the default in the future.  If we notice that the use case changes so much that it deserves to be the default, we can opt to add that step to the installation -- trading extra download time/bandwidth for the convenience of not having to ask for it.

Answering the other question: in my example, the `asset` was meant to be the `avocado.utils.asset` library, but I failed to use the actual API (it was just a mock).  I meant `avocado.utils.asset.Asset` and `avocado.utils.asset.Asset().fetch()`.

Reply:
Got it. I will convert this to a script inside `contrib/scripts/` in a different PR, since it is a new approach.